Core/Utilities/Distribution.py

```diff
@@ -1,464 +1,86 @@
-    return self.executeCommand( cmd, False )
+import re
+import tarfile
+import os
 
-  def doCommit( self, location, comment ):
-    destT = self.__getDevCmdBase( "" )
-    cmd = "svn ci -m '%s' %s '%s'" % ( comment, destT[0], location )
-    return self.executeCommand( cmd, False )
+from DIRAC import S_OK, S_ERROR
+from DIRAC.Core.Utilities import File
 
-  #Get copy revision
-  def getCopyRevision( self, location ):
-    destT = self.__getDevCmdBase( location )
-    cmd = "svn log --stop-on-copy %s '%s'" % ( destT[0], destT[1] )
-    exitCode, outData = self.executeCommand( cmd )
-    if exitCode:
-      return 0
-    copyRev = 0
-    revRE = re.compile( "r([0-9]+)\s*\|\s*(\w+).*" )
-    for line in List.fromChar( outData, "\n" ):
-      reM = revRE.match( line )
-      if reM:
-        copyRev = reM.groups()[0]
-    return copyRev
 
-  #
-  def writeVersionToTmpInit( self, version ):
-    verTup = parseVersionString( version )
-    if not verTup:
-      return False
-    destT = self.__getDevCmdBase( "%s/trunk/%s/__init__.py" % ( self.package, self.package ) )
-    cmd = "svn cat %s '%s'" % ( destT[0], destT[1] )
-    exitCode, outData = self.executeCommand( cmd )
-    if exitCode:
-      return False
-    tmpfd, tmpname = tempfile.mkstemp()
-    versionStrings = ( "majorVersion", "minorVersion", "patchLevel", "preVersion" )
-    reList = []
-    for iP in range( len( versionStrings ) ):
-      if verTup[iP]:
-        replStr = "%s = %s" % ( versionStrings[iP], verTup[iP] )
-      else:
-        replStr = "%s = 0" % versionStrings[iP]
-      reList.append( ( re.compile( r"^(%s\s*=)\s*[0-9]+\s*" % versionStrings[iP] ), replStr ) )
-    for line in outData.split( "\n" ):
-      for reCm, replStr in reList:
-        line = reCm.sub( replStr, line )
-      os.write( tmpfd, "%s\n" % line )
-    os.close( tmpfd )
-    return tmpname
+gVersionRE = re.compile("v([0-9]+)(?:r([0-9]+))?(?:p([0-9]+))?(?:-pre([0-9]+))?")
 
-#End of Distribution class
 
-gVersionRE = re.compile( "v([0-9]+)(?:r([0-9]+))?(?:p([0-9]+))?(?:-pre([0-9]+))?" )
-def parseVersionString( version ):
-  result = gVersionRE.match( version.strip() )
+def parseVersionString(version):
+  result = gVersionRE.match(version.strip())
   if not result:
     return False
   vN = []
   for e in result.groups():
     if e:
-      vN.append( int( e ) )
+      vN.append(int(e))
     else:
-      vN.append( None )
-  return tuple( vN )
+      vN.append(None)
+  return tuple(vN)
 
-def writeVersionToInit( rootPath, version ):
-  verTup = parseVersionString( version )
+
+def writeVersionToInit(rootPath, version):
+  verTup = parseVersionString(version)
   if not verTup:
     return S_OK()
-  initFile = os.path.join( rootPath, "__init__.py" )
-  if not os.path.isfile( initFile ):
+  initFile = os.path.join(rootPath, "__init__.py")
+  if not os.path.isfile(initFile):
     return S_OK()
   try:
-    fd = open( initFile, "r" )
+    fd = open(initFile, "r")
```

Comment:
`with open...`

Reply:
Doesn't the distribution tools need to be python 2.6 compatible ?

Reply:
No, the distribution tools should be OK with 2.7 only.